index.js

```diff
@@ -138,13 +142,23 @@ function createHarness (conf_) {
     var only = false;
     test.only = function (name) {
         if (only) throw new Error('there can only be one only test');
-        results.only(name);
         only = true;
-        return test.apply(null, arguments);
+        var t = test.apply(null, arguments);
+        if (!onlyTestNumber)
+            results.only(t.number);
```

Comment:
please add curly braces around all if blocks, in this case for consistency with lines 107-109.